### PR TITLE
Deprecation warnings for v1beta1 webhook and crd

### LINKED ISSUE #1591: k8s 1.16+ deprecation warnings in development profile
/kind bug

* apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+
* admissionregistration.k8s.io/v1beta1 MutatingWebhookConfiguration is deprecated in v1.16+
* using legacy gatewayName format "knative-ingress-gateway.knative-serving"
* admissionregistration.k8s.io/v1beta1 ValidatingWebhookConfiguration is deprecated in v1.16+

**What steps did you take and what happened:**
`kustomize build config/overlays/development | kubectl apply --validate=false -f -`
```
namespace/kfserving-system created
Warning: apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
customresourcedefinition.apiextensions.k8s.io/inferenceservices.serving.kubeflow.org created
customresourcedefinition.apiextensions.k8s.io/trainedmodels.serving.kubeflow.org created
Warning: admissionregistration.k8s.io/v1beta1 MutatingWebhookConfiguration is deprecated in v1.16+, unavailable in v1.22+; use admissionregistration.k8s.io/v1 MutatingWebhookConfiguration
mutatingwebhookconfiguration.admissionregistration.k8s.io/inferenceservice.serving.kubeflow.org created
serviceaccount/kfserving-models-web-app created
role.rbac.authorization.k8s.io/leader-election-role created
clusterrole.rbac.authorization.k8s.io/kfserving-manager-role created
clusterrole.rbac.authorization.k8s.io/kfserving-models-web-app-cluster-role created
clusterrole.rbac.authorization.k8s.io/kfserving-proxy-role created
rolebinding.rbac.authorization.k8s.io/leader-election-rolebinding created
clusterrolebinding.rbac.authorization.k8s.io/kfserving-manager-rolebinding created
clusterrolebinding.rbac.authorization.k8s.io/kfserving-models-web-app-binding created
clusterrolebinding.rbac.authorization.k8s.io/kfserving-proxy-rolebinding created
configmap/inferenceservice-config created
configmap/kfserving-config created
configmap/kfserving-models-web-app-config-6t9ktmdt9c created
secret/kfserving-webhook-server-secret created
service/kfserving-controller-manager-metrics-service created
service/kfserving-controller-manager-service created
service/kfserving-models-web-app created
service/kfserving-webhook-server-service created
deployment.apps/kfserving-models-web-app created
statefulset.apps/kfserving-controller-manager created
certificate.cert-manager.io/serving-cert created
issuer.cert-manager.io/selfsigned-issuer created
Warning: using legacy gatewayName format "knative-ingress-gateway.knative-serving"; prefer the <namespace>/<name> format: "knative-serving/knative-ingress-gateway"
virtualservice.networking.istio.io/kfserving-models-web-app created
Warning: admissionregistration.k8s.io/v1beta1 ValidatingWebhookConfiguration is deprecated in v1.16+, unavailable in v1.22+; use admissionregistration.k8s.io/v1 ValidatingWebhookConfiguration
validatingwebhookconfiguration.admissionregistration.k8s.io/inferenceservice.serving.kubeflow.org created
validatingwebhookconfiguration.admissionregistration.k8s.io/trainedmodel.serving.kubeflow.org created
```

**What did you expect to happen:**
no warnings

**Environment:**

- Istio Version: 1.8.2
- Knative Version: 0.22
- KFServing Version: master branch
- Kubeflow version: -
- Kubernetes version: (use `kubectl version`): 1.19

### PR DESCRIPTION
**Pull request by andyi2it**
**What this PR does / why we need it**:
Migrates admissionregistration and apiextensions to v1 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1591 

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

File: config/webhook/manifests.yaml [hunk 1]
```diff
@@ -83,6 +89,8 @@ webhooks:
         path: /validate-serving-kserve-io-v1alpha2-inferenceservice
     failurePolicy: Fail
     name: inferenceservice.kserve-webhook-server.validator
+    sideEffects: None
+    admissionReviewVersions: ["v1beta1"]
```
Comment (yuzisun) [inline config/webhook/manifests.yaml]: here should be v1alpha2?

File: config/webhook/manifests.yaml [hunk 2]
```diff
@@ -126,6 +136,8 @@ webhooks:
         path: /validate-serving-kserve-io-v1alpha1-trainedmodel
     failurePolicy: Fail
     name: trainedmodel.kserve-webhook-server.validator
+    sideEffects: None
+    admissionReviewVersions: ["v1beta1"]
```
Comment (yuzisun) [inline config/webhook/manifests.yaml:100]: this should be v1alpha1?

File: config/webhook/manifests.yaml [hunk 3]
```diff
@@ -10,14 +10,16 @@ webhooks:
       service:
         name: $(webhookServiceName)
         namespace: $(kserveNamespace)
-        path: /mutate-serving-kserve-io-v1alpha2-inferenceservice
+        path: /mutate-serving-kserve-io-v1beta1-inferenceservice
```
Comment (yuzisun) [inline config/webhook/manifests.yaml:13]: this looks like a duplicate of the following `clientConfig`.

File: config/webhook/manifests.yaml (hunk 1 above)
Comment (andyi2it) [reply config/webhook/manifests.yaml]: `v1` version wouldn't allow `v1alpha2` for admissionReviewVersions.

Comment (andyi2it) [reply config/webhook/manifests.yaml]: supported versions are `v1` and `v1beta1`.

Comment (yuzisun) [reply config/webhook/manifests.yaml]: sorry I misunderstood this field, I thought it is the custom resource version

**Comment by yuzisun** [comment]
Thanks @andyi2it !
/lgtm
/approve

**Pull request merged**